File: 프로그래머스/lv2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        Queue<Integer> queue = new LinkedList<>(); 
    ArrayList<Integer> list = new ArrayList<>();
    for(int i=0; i<progresses.length; i++){
      double re = (100-progresses[i])/(double) speeds[i];
      int day = (int) Math.ceil(re);

      if(!queue.isEmpty() && queue.peek() < day){
        list.add(queue.size());
        queue.clear();
      }

      queue.add(day);
    }
    list.add(queue.size());

    int[] answer = new int[list.size()];
    for(int i=0; i<list.size(); i++){
      answer[i] = list.get(i);
    }
        
        
        return answer;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    const answer = [];
    
    const N = progresses.length;
    const days = progresses.map((progress, i) => Math.ceil((100 - progress) / speeds[i]));

    let prev = days[0];
    let count = 1;
    for (let i = 1; i < N; i++) {
        const target = days[i];
        
        if (prev >= target) {
            count++;
        }else {
            answer.push(count);
            prev = target;
            count = 1;
        }
    }
    answer.push(count);
    
    return answer;
}